index.bs

```diff
@@ -866,6 +866,44 @@ it is not always *sufficient*
 to protect users from invasive behaviours,
 and seeking [meaningful consent](#consent) is also important.
 
+<h3 id="handle-destroyed-contexts">Handle non-fully-active documents (and destroyed execution contexts)</h3>
+
+When an <{iframe}> (or similar) is removed or navigated,
+its document stops being [=Document/fully active=]
+and its [=ECMAScript/execution contexts=] can get destroyed.
+However, references to objects from that document
+can remain reachable from other documents
+or from other realms.
+
+Specifications should say what happens both at call time
+and after an operation has begun:
+otherwise background work may keep running after a page is gone,
+UI can hang, and user agents diverge in abort/cleanup behavior.
+
+If the [=relevant global object=]'s [=associated Document=]
+is not [=Document/fully active=] at call time,
+then [=throw=], or return [=a promise rejected with=],
+with {{InvalidStateError}}.
+
+If, after starting an asynchronous operation,
+that [=associated Document=] becomes non-[=Document/fully active=]
+or the [=relevant global object=]'s [=ECMAScript/execution context=] is destroyed by script,
+abort the operation and reject with {{AbortError}}.
```

Comment:
This doesn't seem to always be the right advice. E.g. https://html.spec.whatwg.org/multipage/images.html#update-the-image-data just waits for the document to become fully active again.